### ISSUE TITLE
[Bug]: 11.1.5-SNAPSHOT.20240920041432: None of my CUEsheets seem to be being indexed!

### ISSUE THREAD

**Issue by Beechwoods** [kagemomiji/airsonic-advanced#635]
### What happened?

My collection is entirely indexed using CUEsheets, to ensure that all Metadata is consistent. Some CUEsheets index single file FLAC's and WAV's, some index multi-file sets. When I switch on Enable cue indexing, none of the CUE's seem to be indexed and the log is filled with warnings:

2024-11-16 19:11:33.206 WARN --- o.a.p.service.MediaFileService : Error parsing cuesheet /Users/#####/Music/Subsonic Library/Official/Lossless/Bill Evans/Bandleader/Bill Evans/(1968) - At the Montreux Jazz Festival/1998 - At the Montreux Jazz Festival (Verve Records, MGM Records 539 758-2, V6-8762, MGS-1707, Europe, CD, Remastered)/!.flac.cue
--
2024-11-16 19:11:33.401 WARN --- o.a.p.service.MediaFileService : Error parsing cuesheet /Users/#####/Music/Subsonic Library/Official/Lossless/The Crusaders/Various Artists/(1994) - The Crusaders and Beyond/1994 - The Crusaders and Beyond... (Music Club MCCD 163, UK, CD, Compilation)/The Crusaders and Beyond_Noncompliant.cue
2024-11-16 19:11:33.518 WARN --- o.a.p.service.MediaFileService : Error parsing cuesheet /Users/#####/Music/Subsonic Library/Official/Lossless/+ CCM/Wolves at the Gate/(2016) - Types & Shadows/2016 - Types & Shadows (Solid State TND000060, 810488022445, US, CD, Album)/Types & Shadows.cue
2024-11-16 19:11:33.545 WARN --- o.a.p.service.MediaFileService : Error parsing cuesheet /Users/#####/Music/Subsonic Library/Official/Lossless/McCarthy/(2014) - Complete Albums, Singles And BBC Sessions Collection (FLAC) CRCDMBOX15/Compiled B-Sides/(1988) - This Nelson Rockefeller 12'/!.flac.cue

Once indexing is complete, I only see the original Files, not the indexed CUE tracks, in the UI.

All CUE files work fine in Foobar2000 and MPD.

I am new to Airsonic Advanced and this branch (which I was keen on because of it's CUE indexing functionality) so do not know if this is a new issue or not.

### Steps to reproduce

1. Select Enable cue indexing
2. Click 'Scan Media Folders now'
3. Review logs and UI once indexing is complete.

### Version

11.1.5-SNAPSHOT (Edge)

### Version Detail

11.1.5-SNAPSHOT.20240920041432

### Operating System

MacOS Monterey 12.7.6

### Java Version

java version "23.0.1" 2024-10-15 Java(TM) SE Runtime Environment (build 23.0.1+11-39) Java HotSpot(TM) 64-Bit Server VM (build 23.0.1+11-39, mixed mode, sharing)

### Database

Other

### DB Detail

Built-in

### Configuration paramter

JWTKey=j7jghc3vt6tcm2v8kmltf3ueup
SettingsChanged=1731783682084
EncryptionKeyPassword=############################
EncryptionKeySalt=############################
GettingStartedEnabled=false
IndexCreationInterval=1
IndexCreationHour=3
FastCacheEnabled=false
OrganizeByFolderStructure=true
UploadsFolder=%{['USER_MUSIC_FOLDERS'][0]}/Incoming
ExcludePattern=
IgnoreSymLinks=false
HideIndexedFiles=true
EnableCueIndexing=true
SplitOptions=-ss %o -t %d
SplitCommand=ffmpeg %S -i %s -vcodec copy -acodec copy -f %f -
DownsamplingCommand4=ffmpeg %S -i %s -map 0:0 -b:a %bk -v 0 -f mp3 -
HlsCommand4=ffmpeg -ss %o -i %s -s %wx%h -async 1 -c:v libx264 -flags +cgop -b:v %vk -maxrate %bk -preset superfast -copyts -b:a %rk -bufsize 256k -map 0:0 -map 0:%i -ac 2 -ar 44100 -v 0 -threads 0 -force_key_frames expr:gte(t,n_forced*10) -start_number %j -hls_time %d -hls_list_size 0 -hls_segment_filename %n %p
JukeboxCommand2=ffmpeg -ss %o -i %s -map 0:0 -v 0 -ar 44100 -ac 2 -f s16be -
VideoImageCommand=ffmpeg -r 1 -ss %o -t 1 -i %s -s %wx%h -v 0 -f mjpeg -
SubtitlesExtractionCommand=ffmpeg -i %s -map 0:%i -f %f -
TranscodeEstimateTimePadding=2000
TranscodeEstimateBytePadding=0
FullScan=false
ClearFullScanSettingAfterScan=false

### Proxy Server

None

### client detail

Google Chrome Version 130.0.6723.119 (Official Build) (x86_64)

### language

English (United Kingdom)

### Relevant log output

```console
2024-11-16 19:13:36.687  WARN --- o.a.p.service.MediaFileService           : Error parsing cuesheet /Users/#####/Music/Subsonic Library/Official/Lossless/Giorgio Moroder/Donna Summer/(1977) - I Feel Love - Theme from the Deep (Down, Deep Inside)/1977 - I Feel Love - Theme from the Deep (Down, Deep Inside) (Casablanca NBD 20104, US, Vinyl, Pitman Pressing Plant)/!.flac.cue
2024-11-16 19:13:36.767  WARN --- o.a.p.service.MediaFileService           : Error parsing cuesheet /Users/#####/Music/Subsonic Library/Official/Lossless/Giorgio Moroder/Giorgio Moroder/(1978) - Chase/1978 - Chase (Casablanca NBD 20146 DJ, US, Vinyl, Promo)/!.flac.cue
2024-11-16 19:13:36.798  WARN --- o.a.p.service.MediaFileService           : Error parsing cuesheet /Users/#####/Music/Subsonic Library/Official/Lossless/Porcupine Tree/Compilations, Promos/(2001) - Metanoia (FLAC) Delerium DELEC CD 079, UK Issue/!.flac.cue
```

**Comment by yuygfgg** [comment]
only utf8 without bom is supported currently

**Comment by UVJkiNTQ** [comment]
#624 https://github.com/kagemomiji/airsonic-advanced/issues/624

**Comment by Beechwoods** [comment]
Interestingly, I downloaded the Test CUE and File sets from the src/resources area of the Source Code in the repo, and none of these are sucessfully processed by my installation either. All I see is the base files. 

`/private/var/airsonic/airsonic.log:18939: 2024-11-17 12:36:10.260  WARN --- o.a.p.service.MediaFileService           : Could not find base file 'airsonic-test.flac' for cue sheet General Audio/Airsonic-Test/airsonic-test.cue
/private/var/airsonic/airsonic.log:19122: 2024-11-17 12:37:40.463  WARN --- o.a.p.service.MediaFileService           : Could not find base file 'airsonic-test.flac' for cue sheet General Audio/Airsonic-Test/airsonic-test.cue
/private/var/airsonic/airsonic.log:19216: 2024-11-17 12:39:27.327  WARN --- o.a.p.service.MediaFileService           : Could not find base file 'airsonic-test.flac' for cue sheet General Audio/Airsonic-Test/airsonic-test.cue
/private/var/airsonic/airsonic.log:19345: 2024-11-17 12:40:28.021  WARN --- o.a.p.service.MediaFileService           : Could not find base file 'airsonic-test.flac' for cue sheet Airsonic-Test/airsonic-test.cue
/private/var/airsonic/airsonic.log:19353: 2024-11-17 12:47:17.568  WARN --- o.a.p.service.MediaFileService           : Error parsing cuesheet /Users/##########/Music/Subsonic Library/Incoming/Airsonic-Test/disableCue/airsonic-test.cue
/private/var/airsonic/airsonic.log:19354: 2024-11-17 12:47:17.593  WARN --- o.a.p.service.MediaFileService           : Error parsing cuesheet /Users/##########/Music/Subsonic Library/Incoming/Airsonic-Test/invalidCue3/airsonic-test.cue
/private/var/airsonic/airsonic.log:19362: 2024-11-17 12:47:17.823  WARN --- o.a.p.service.MediaFileService           : Could not find base file 'airsonic-test.flac' for cue sheet Airsonic-Test/cueAndFlac/airsonic-test.cue
/private/var/airsonic/airsonic.log:19569: 2024-11-17 12:50:08.261  WARN --- o.a.p.service.MediaFileService           : Could not find base file 'airsonic-test.flac' for cue sheet Airsonic-Test/cueAndFlac/airsonic-test.cue
/private/var/airsonic/airsonic.log:19570: 2024-11-17 12:50:22.807  WARN --- o.a.p.service.MediaFileService           : Error parsing cuesheet /Users/##########/Music/Subsonic Library/Incoming/Airsonic-Test/invalidCue3/airsonic-test.cue
/private/var/airsonic/airsonic.log:19586: 2024-11-17 12:54:17.732  WARN --- o.a.p.service.MediaFileService           : Could not find base file 'airsonic-test.flac' for cue sheet Airsonic-Test/cueAndFlac/airsonic-test.cue
/private/var/airsonic/airsonic.log:19587: 2024-11-17 12:54:20.850  WARN --- o.a.p.service.MediaFileService           : Error parsing cuesheet /Users/##########/Music/Subsonic Library/Incoming/Airsonic-Test/disableCue/airsonic-test.cue
/private/var/airsonic/airsonic.log:19588: 2024-11-17 12:54:32.717  WARN --- o.a.p.service.MediaFileService           : Error parsing cuesheet /Users/##########/Music/Subsonic Library/Incoming/Airsonic-Test/invalidCue3/airsonic-test.cue
`
Here is a screenshot of the directory structure for these files:
![Screenshot 2024-11-17 at 12 57 51](https://github.com/user-attachments/assets/a8d7d3cc-7eb4-4962-b6fc-94d38f8fce93)

**Comment by Beechwoods** [comment]
I have been able to get the test CUE files included in the Source Code working, by unchecking 'Hide Virtual Tracks' - from the tooltip it looked like ticking this would hide the actual tracks created by the CUE file, but it seems to do the opposite. Now checking if I can see my own CUE file tracks. It looks like some at least are missing. 

The example above, is now working. I see the CUE indexed files, plus the original, untagged FLAC files.

**Comment by Beechwoods** [comment]
This is how that set looks with 'Hide Virtual Tracks' unchecked.
![Screenshot 2024-11-17 at 13 28 52](https://github.com/user-attachments/assets/8376cb99-1f4a-4a52-9fa1-c79018b71995)
